### PR TITLE
first pass at vnx support

### PR DESCRIPTION
**Pull request by paulczar**
documentation: http://docs.openstack.org/juno/config-reference/content/emc-vnx-direct-driver.html

some steps will still be manual as we can't test them without a real vnx
  8 see `Steps for EMCCLIISCSIDriver`

### CODE REVIEW

**Comment by craigtracey** [comment]
:LGTM: 

@dlundquist thoughts?

File: roles/cinder-common/templates/etc/cinder/cinder.conf [hunk 1]
```diff
@@ -66,11 +66,22 @@ volume_clear_size = {{ cinder.volume_clear_size }}
...
 san_login = {{ backend.san_login }}
 san_password = {{ backend.san_password }}
 {% if backend.nimble_subnet_label is defined %}
 nimble_subnet_label = {{ backend.nimble_subnet_label }}
+{% endif -%}
+{% if backend.name == 'vnx' -%}
+storage_vnx_pool_name = {{ backend.pool_name }}
+naviseccli_path = {{ backend.naviseccli_path }}
+storage_vnx_authentication_type = {{ backend.authentication_type }}
+initiator_auto_registration = True
```
Comment (dlundquist) [inline roles/cinder-common/templates/etc/cinder/cinder.conf]: Are these VNX specific? This seems to be getting a little complicated with conditionals, does Jinja have the concept of partials?

File: roles/cinder-data/tasks/main.yml [hunk 2]
```diff
@@ -39,6 +39,10 @@
 - name: iscsi start
   service: name=open-iscsi state=started
 
+- include: vnx.yml
```
Comment (dlundquist) [inline roles/cinder-data/tasks/main.yml]: Maybe we should call this vnx_navicli.yml?

**Comment by dlundquist** [comment]
@paulczar Are you intending to incorporate my comments above, or do we merge as is?

**Comment by paulczar** [comment]
Yeah, I want to get it working first, then we can cleanup and figure out how/where to merge.

File: roles/cinder-data/tasks/main.yml (hunk 2 above)
Comment (paulczar) [inline roles/cinder-data/tasks/main.yml]: this is starting to do more than just the navicli,  multipath and other stuff.   maybe we could call it `vnx_backend.yml` or something like that if `vnx` isn't descriptive enough.

File: roles/cinder-common/templates/etc/cinder/cinder.conf (hunk 1 above)
Comment (paulczar) [inline roles/cinder-common/templates/etc/cinder/cinder.conf]: I can dig into jinja partials ... or I think we could use a `cinder.d` style directory and give each backend its own template / config file ?

Comment (dlundquist) [inline roles/cinder-common/templates/etc/cinder/cinder.conf]: Sourcing multiple configurations files from /etc/cinder sounds like a good way to go.

File: roles/cinder-data/tasks/main.yml (hunk 2 above)
Comment (dlundquist) [inline roles/cinder-data/tasks/main.yml]: Many backends support multipath, that doesn't seem to be VNX specific, and it seems like we could include that in the core role, or perhaps a common include for backends which support it.

File: roles/cinder-common/templates/etc/cinder/cinder.conf (hunk 1 above)
Comment (paulczar) [inline roles/cinder-common/templates/etc/cinder/cinder.conf]: I think this is doable,  But I want to punt on it until we decide we want to merge this support into master itself at which point we can rework it.  I don't want to spend much  time on something we might not actually mainline.

**Pull request closed without merging**